File: ShadowFrameworkLite20/src/shadow/pipeline/SFPipelineTransform3f.java
package shadow.pipeline;

import shadow.math.SFMatrix3f;
import shadow.math.SFVertex3f;

public class SFPipelineTransform3f {

	private SFRigidTransform3fArray array;
	private int index;
	
	public SFPipelineTransform3f(SFRigidTransform3fArray array, int index) {
		super();
		this.array = array;
		this.index = index;
	}
	
	public void attachOn(SFPipelineTransform3f transform){
		transform.array.attach(array, index, transform.index);
	}
	
	public void setPosition(SFVertex3f vertex){
		array.setElementPosition(index, vertex);
	}
	
	public void setOrientation(SFMatrix3f matrix){
		array.setElementOrientation(index, matrix);
	}

	public void getPosition(SFVertex3f vertex){
		array.getElementPosition(index, vertex);
	}
	
	public void getOrientation(SFMatrix3f matrix){
		array.getElementOrientation(index, matrix);
	}
	
	public void apply(){
		array.apply(index);
	}
}

File: ShadowFrameworkJS/src/shadow/pipeline/SFPipelineTransform3f.js

function SFPipelineTransform3f(array,index){
	this.array = array;
	this.index = index;
}

SFPipelineTransform3f.prototype["attachOn"]=function(transform){
	transform.array.attach(this.array, this.index, transform.index);
};

SFPipelineTransform3f.prototype["setPosition"]=function(vertex){
	this.array.setElementPosition(this.index, vertex);
};

SFPipelineTransform3f.prototype["setOrientation"]=function(matrix){
	this.array.setElementOrientation(this.index, matrix);
};

SFPipelineTransform3f.prototype["getPosition"]=function(vertex){
	this.array.getElementPosition(this.index, vertex);
};

SFPipelineTransform3f.prototype["getOrientation"]=function(matrix){
	this.array.getElementOrientation(this.index, matrix);
};

SFPipelineTransform3f.prototype["apply"]=function(){
	this.array.apply(this.index);
};	